src/gui/editors/AutomationEditor.cpp

```diff
@@ -1867,33 +1872,37 @@ AutomationEditor::timeMap::iterator AutomationEditor::getNodeAt(int x, int y, bo
 	timeMap & tm = m_pattern->getTimeMap();
 	timeMap::iterator it = tm.begin();
 
-	// If there's a node that is inside those coordinates we will store it here
-	timeMap::iterator node = tm.end();
-
 	// ticksOffset is the number of ticks that match "r" pixels
 	int ticksOffset = MidiTime::ticksPerBar() * r / m_ppb;
 
 	while (it != tm.end())
 	{
 		// If the x coordinate is within "r" pixels of the node's position
-		if (posTicks >= it.key() - ticksOffset && posTicks <= it.key() + ticksOffset)
+		if (posTicks >= it.key() - ticksOffset)
 		{
-			// The y position of the node
-			float valueY = yCoordOfLevel(
-				outValue
-					? it.value().getOutValue()
-					: it.value().getInValue()
-				);
-			// If the y coordinate is within "r" pixels of the node's value
-			if (y >= (valueY - r) && y <= (valueY + r))
+			if (posTicks <= it.key() + ticksOffset)
```

Comment:
Is this change to improve readability or for another reason? I don't see why you've split up the conditional.

Reply:
I moved it down so I can break out of the loop if the X coordinate is lower then `it.key() - ticksOffset` for a particular node. Since as @Spekular observed, the nodes are all ordered in the timeMap. So if the position is lower than `it.key()-ticksOffset` for one node, it will also be for all subsequent ones. That saves some unnecessary loop iterations in some situations. Maybe I can come up with some comments that make that a little bit more clear

Reply:
Gotcha, I think I missed the `else break;` part since GitHub tabs differ from file to file for some reason for me. lol